### DISCUSSION TITLE
Unable to change MapTypeId

### DISCUSSION THREAD

**Discussion by pierregremaud** [inocan-group/vue3-google-map#213] [Q&A] [answered]
Looks like property "mapTypeId" is not working in component GoogleMap

My component template looks like : 

            <GoogleMap
                api-key="xxxxxxxxxxxxxxx" 
                style="width: 100%; height: 600px"
                :mapTypeId="terrain"
            >

But the mapTypeId is allways the default ("roadmap")

**Comment by JoseGoncalves** [comment, marked as answer]
You need to pass the parameters in the hyphenated form. Try:

```html
        <GoogleMap
            api-key="xxxxxxxxxxxxxxx" 
            style="width: 100%; height: 600px"
            :map-type-id="terrain"
        >
```

**Comment by JoseGoncalves** [comment]
If `terrain` is a String instead of a Vue Ref, you need to set it like this:
```html
        <GoogleMap
            api-key="xxxxxxxxxxxxxxx" 
            style="width: 100%; height: 600px"
            map-type-id="terrain"
        >
```

**Comment by pierregremaud** [comment]
Yesss !!! Your solution works perfectely for me.

Many thanks for your help and for giving us that great software.

**Comment by JoseGoncalves** [comment]
@pierregremaud You're welcome but, I did not develop the software, I just use it and made some minor contributions.

**Comment by HusamElbashir** [comment]
@JoseGoncalves Is a regular contributor to this library and often steps in to answer questions when us maintainers are busy. Much appreciated chief 🙏
